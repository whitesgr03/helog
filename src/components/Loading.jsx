import { useContext } from "react";

import style from "../styles/Loading.module.css";
import image from "../styles/utils/image.module.css";

import { DarkThemeContext } from "../contexts/DarkThemeContext";

const Loading = () => {
	const [darkTheme] = useContext(DarkThemeContext);
	return (
		<div className={`${darkTheme ? style.dark : ""} ${style.loading}`}>
			<span className={`${image.icon} ${style.load}`} />
			Loading ...
		</div>
	);
};

export default Loading;
