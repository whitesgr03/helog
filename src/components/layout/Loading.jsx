import PropTypes from "prop-types";

// Styles
import style from "../../styles/layout/Loading.module.css";
import image from "../../styles/utils/image.module.css";

export const Loading = ({ text, dark, light, shadow }) => {
	return (
		<div
			className={`${style.loading} ${dark ? style.dark : ""} ${
				light ? style.light : ""
			} ${shadow ? style.shadow : ""}`}
		>
			<span className={`${image.icon} ${style.load}`} />
			{text}
		</div>
	);
};

Loading.propTypes = {
	text: PropTypes.string,
	dark: PropTypes.bool,
	light: PropTypes.bool,
	shadow: PropTypes.bool,
};
