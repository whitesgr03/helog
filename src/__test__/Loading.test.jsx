import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";

import Loading from "../components/Loading";

describe("Error component", () => {
	it("should render correctly", () => {
		const { asFragment } = render(<Loading />);

		const actual = asFragment();

		expect(actual).toMatchSnapshot();
	});
});
