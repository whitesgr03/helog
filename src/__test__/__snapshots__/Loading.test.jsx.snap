// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading component > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_loading_87501d"
  >
    <span
      class="_icon_7e06e0 _load_87501d"
    />
    Loading ...
  </div>
</DocumentFragment>
`;
